### PR TITLE
Improve CI (coordicide)

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

Improve the performance of CI, analogous to #856 on `dev`.

## Links to any relevant issues

fixes issue #854 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran CI.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Comment by thibault-martinez** [comment]
> ## Pull Request Test Coverage Report for [Build 1535247850](https://coveralls.io/builds/44740303)
> * **0** of **0**   changed or added relevant lines in **0** files are covered.
> * No unchanged relevant lines lost coverage.
> * Overall coverage increased (+**0.1%**) to **75.735%**
> 
> Totals	[Coverage Status](https://coveralls.io/builds/44740303)
> Change from base [Build 1531565737](https://coveralls.io/builds/44715356):	0.1%
> Covered Lines:	3271
> Relevant Lines:	4319
> ##### 💛 - [Coveralls](https://coveralls.io)

This is new

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+      - name: Cache cargo state
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-cargo
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            ~/.cargo/bin
+          key: ${{ matrix.os }}-${{ matrix.rust }}-${{ env.cache-name }}-${{ hashFiles('**/Cargo.toml') }}-build
```
Comment (thibault-martinez) [inline .github/workflows/ci.yml]: There is no `${{ matrix.os }}` for this one ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Can you put missing files without requirement as it runs in < 5s please ?

**Comment by grtlr** [comment]
Sure, no problem.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**